### PR TITLE
Adding the information about the danger of tx.origin in the recommend…

### PR DESCRIPTION
**Pull request by yippee-ki-yay**
#1 
I can provide an example of an attack if you use tx.origin, but it's already in the solidity docs so I just linked that.
If you want me to add anything else let me know.

### CODE REVIEW

File: docs/recommendations.md [hunk 1]
```diff
@@ -308,4 +308,14 @@ These are recommendations which are no longer relevant due to changes in the pro
 
 ### Beware division by zero (Solidity < 0.4)
 
-Prior to version 0.4, Solidity [returns zero](https://github.com/ethereum/solidity/issues/670) and does not `throw` an exception when a number is divided by zero. Ensure you're running at least version 0.4.
\ No newline at end of file
+Prior to version 0.4, Solidity [returns zero](https://github.com/ethereum/solidity/issues/670) and does not `throw` an exception when a number is divided by zero. Ensure you're running at least version 0.4.
+
+### Avoid using tx.origin
```
Comment (maurelian) [inline docs/recommendations.md]: Please move this section above the `Deprecated/historical recommendations`. Would appreciate if you could also change that section header to an h2 (##) while you're at it.

File: docs/recommendations.md [hunk 2]
```diff
@@ -308,4 +308,14 @@ These are recommendations which are no longer relevant due to changes in the pro
...
+
+Never use tx.origin for authorization, another contract can have a method which will call your contract (where the user has some funds for instance) and your contract will authorize that transaction as your address is in tx.origin.
+
+You can read more about it here: [Solidity docs](https://solidity.readthedocs.io/en/develop/security-considerations.html#tx-origin)
+
+Besides the issue with authorization, there is a chance that tx.origin will be removed from the Ethereum protocol in the future, so code that uses tx.origin won't be compatible with future releases [Vitalik: 'Do NOT assume that tx.origin will continue to be usable or meaningful.'](https://ethereum.stackexchange.com/questions/196/how-do-i-make-my-dapp-serenity-proof/200#200)
+
+It's also worth mentioning that tx.origin means that your contract that uses tx.origin cannot be used by another contract as a contract can't be the tx.origin.
```
Comment (maurelian) [inline docs/recommendations.md]: Please put the word "interoperability" in here somewhere.

File: docs/recommendations.md [hunk 3]
```diff
@@ -308,4 +308,14 @@ These are recommendations which are no longer relevant due to changes in the pro
 
 ### Beware division by zero (Solidity < 0.4)
 
-Prior to version 0.4, Solidity [returns zero](https://github.com/ethereum/solidity/issues/670) and does not `throw` an exception when a number is divided by zero. Ensure you're running at least version 0.4.
\ No newline at end of file
+Prior to version 0.4, Solidity [returns zero](https://github.com/ethereum/solidity/issues/670) and does not `throw` an exception when a number is divided by zero. Ensure you're running at least version 0.4.
+
+### Avoid using tx.origin
+
+Never use tx.origin for authorization, another contract can have a method which will call your contract (where the user has some funds for instance) and your contract will authorize that transaction as your address is in tx.origin.
```
Comment (maurelian) [inline docs/recommendations.md]: please apply inline code formatting to `tx.origin` wherever it appears. 

Also, you should explain that `msg.sender` is the preferred alternative.

**Comment by yippee-ki-yay** [comment]
Hi @maurelian thanks for the feedback.
I've changed the stuff you asked, let me know if there's anything else 👍

**Comment by yippee-ki-yay** [comment]
@maurelian Ok sorry about that, I've added an example of the attack pretty much the same as the docs. But thats the gist of it without getting to complicated.

**Comment by maurelian** [comment]
Excellent, thank you!

**Comment by yippee-ki-yay** [comment]
@maurelian No problem happy to help out, can the issue be closed now, I'm now sure how to get the bounty of gitcoin?

**Comment by maurelian** [comment]
Right, I just released the bounty. TY

**Pull request merged**